### PR TITLE
Set `inputs` as kwarg in `TextClassificationPipeline`

### PR DESCRIPTION
**Pull request by alvarobartt**
# What does this PR do?

This PR replaces the current `*args` arg within `TextClassificationPipeline` with a kwarg named `inputs` to be able to use it as a keyword arg when e.g. trying to unpack a Python dict as `pipeline(**{"inputs": "text"})` which is not possible with the current approach since the `*args` were being used instead.

Additionally, the usage of a kwarg aligns the `TextClassificationPipeline` implementation with the rest of the `Pipeline` subclasses, and not being a breaking change, since before no keyword arg could be specified.

## Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests? -> There was no need to write any test, since the unpacking of `*inputs` works the same way as `*args` did, only adding the possibility to use it as a kwarg too.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Library:

- pipelines: @Rocketknight1

### CODE REVIEW

File: src/transformers/pipelines/text_classification.py [hunk 1]
```diff
@@ -152,10 +152,16 @@ def __call__(self, *args, **kwargs):
 
             If `top_k` is used, one such dictionary is returned per label.
         """
-        result = super().__call__(*args, **kwargs)
+        if isinstance(inputs, str) or isinstance(inputs, dict):
+            inputs = (inputs,)
+        else:
+            inputs = tuple(
+                [inputs],
+            )  # noqa: C409
```
Comment (Rocketknight1) [inline src/transformers/pipelines/text_classification.py:160]: If `inputs` is a list, isn't this equivalent to just `(inputs,)` as well?

**Review by Rocketknight1 — APPROVED** [review]
This seems like a good PR to me (with one formatting nit), and I agree that using `*args` for the sole argument is limiting. We've seen other issues where users want to pass extra keyword args to `TextClassificationPipeline`, such as #29271 - do you think this PR could resolve that issue as well?

File: src/transformers/pipelines/text_classification.py (hunk 1 above)
Comment (alvarobartt) [reply src/transformers/pipelines/text_classification.py:160]: So the issue here is that Python will convert the inner list into a tuple, meaning that `[1, 2, 3]` as `tuple([1, 2, 3],)` is not `([1, 2, 3],)` but `(1, 2, 3)` which is wrong, and in this case just the first arg from `inputs` (the former `*args`) is used, while the rest are discarded and printed out as "Ignored args: (2, 3)"

Comment (alvarobartt) [reply src/transformers/pipelines/text_classification.py:160]: Maybe there's a better workaround formatting-wise to do that in a cleaner way? But I was unsure so I went with the safest approach to keep the current implementation using `*args` as is, with no breaking changes at all.

**Comment by alvarobartt** [comment]
> This seems like a good PR to me (with one formatting nit), and I agree that using `*args` for the sole argument is limiting. We've seen other issues where users want to pass extra keyword args to `TextClassificationPipeline`, such as #29271 - do you think this PR could resolve that issue as well?

Hmm I think that issue is referring to the fact that the input args under `**kwargs` are being "sanitized" so that just the recognized ones are used, while the `forward` method may accept more than those for their specific use case, so not sure this will be covered anyway with this PR, but maybe the `_sanitize_parameters` function could be refactored to validate the "known" ones, while leaving the rest of those there up to the users awareness and propagate those to the `forward` pass of that model? Not sure though, probably could break some UX.

File: src/transformers/pipelines/text_classification.py (hunk 1 above)
Comment (Rocketknight1) [reply src/transformers/pipelines/text_classification.py:160]: Hm - are you sure? This is what I got:
```python
>>> inputs = [1, 2, 3]
>>> tuple([inputs],)
([1, 2, 3],)
>>> (inputs,)
([1, 2, 3],)
```

It looks like both approaches have the same output!

Comment (Rocketknight1) [reply src/transformers/pipelines/text_classification.py:160]: Ah, I see - they differ if `inputs` is a nested list.

Comment (alvarobartt) [reply src/transformers/pipelines/text_classification.py:160]: True, I think that the `tuple([...],)` was what was confusing me, sorry for the misunderstanding, and thanks for the heads up!

**Comment by Rocketknight1** [comment]
@alvarobartt makes sense! Either way, I'm happy to approve this. cc @ArthurZucker for core maintainer review!

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding!

**Review by amyeroberts — APPROVED** [review]

**Pull request merged**